microprofile/cdi/src/main/java/io/helidon/microprofile/cdi/BuildTimeEvent.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (c) 2020 Oracle and/or its affiliates. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.helidon.microprofile.cdi;
+
+import java.lang.annotation.Annotation;
+import java.lang.annotation.Documented;
+import java.lang.annotation.Inherited;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+
+import javax.inject.Qualifier;
+
+/**
+ * An event sent for build time.
+ */
+public class BuildTimeEvent {
+    private final String id;
+
+    BuildTimeEvent(String id) {
+        this.id = id;
+    }
+
+    /**
+     * Container id.
+     *
+     * @return id of this container
+     */
+    public String id() {
+        return id;
+    }
+
+    /**
+     * Build time of the application (as opposed to runtime).
+     *
+     * The initialization happens as soon as possible within CDI bootstrap.
+     */
+    @Retention(RetentionPolicy.RUNTIME)
+    @Documented
+    @Inherited
+    @Qualifier
+    public @interface BuildTimeStart {
+        /**
+         * A literal for this annotation.
+         */
+        BuildTimeStart LITERAL = new BuildTimeStart() {
+            @Override
+            public Class<? extends Annotation> annotationType() {
+                return BuildTimeStart.class;
+            }
+        };
+    }
+
+    /**
+     * End of build time of the application (as opposed to runtime).
+     *
+     * The event is fired when initialization is done. In case of native-image, this would be before
+     *  the native image generation starts.
+     */
+    @Retention(RetentionPolicy.RUNTIME)
+    @Documented
+    @Inherited
+    @Qualifier
+    public @interface BuildTimeEnd {
+        /**
+         * A literal for this annotation.
+         */
+        BuildTimeEnd LITERAL = new BuildTimeEnd() {
```

Comment:
Please subclass [`javax.enterprise.util.AnnotationLiteral`](https://jakarta.ee/specifications/cdi/2.0/apidocs/javax/enterprise/util/AnnotationLiteral.html) for creating annotation literals, following its documentation.

Reply:
Fixed